pinot-controller/src/main/java/com/linkedin/pinot/controller/helix/core/realtime/PinotLLCRealtimeSegmentManager.java

```diff
@@ -639,13 +681,23 @@ public boolean commitSegmentMetadata(String rawTableName, final String committin
     LLCSegmentName newLLCSegmentName = new LLCSegmentName(committingLLCSegmentName.getTableName(), partitionId, newSeqNum, now);
     String newSegmentNameStr = newLLCSegmentName.getSegmentName();
 
-    final PartitionAssignment partitionAssignment = getStreamPartitionAssignment(realtimeTableName);
+    IdealState idealState = getTableIdealState(realtimeTableName);
+    int numPartitions = _partitionAssignmentGenerator.getNumPartitionsFromIdealState(idealState);
```

Comment:
you can add an assert here that the segment that is completing is in CONSUMING state.